Traceback (most recent call last):
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# 每次新打开 notebook 或者重启 kernel 后，该模块均需要执行一次
%matplotlib widget
%gui asyncio

from ai2_kit.feat import catalysis as ai2cat
from ai2_kit.feat.catalysis import ui
import os

AI4EC_DIR              = '/public/groups/ai4ec/data'
CP2K_DATA              = f'{AI4EC_DIR}/cp2k'
CP2K_BATCH_TEMPLATE    = f'{AI4EC_DIR}/slurm/ai2cat-cp2k.sbatch'
LAMMPS_BATCH_TEMPLATE  = f'{AI4EC_DIR}/slurm/ai2cat-lammps.sbatch'
AI2KIT_EXECUTOR_CONFIG = f'{AI4EC_DIR}/ai2-kit/executor.yml'

CP2K_DEFAULTS = dict(
    basis_set_file = os.path.join(CP2K_DATA, 'BASIS_MOLOPT'),
    potential_file = os.path.join(CP2K_DATA, 'GTH_POTENTIALS'),
    parameter_file = os.path.join(CP2K_DATA, 'dftd3.dat'),
)

ui_helper = ui.get_the_ui_helper()
print("Current Directory Is: ", os.getcwd())
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[1], line 2[0m
[1;32m      1[0m [38;5;66;03m# 每次新打开 notebook 或者重启 kernel 后，该模块均需要执行一次[39;00m
[0;32m----> 2[0m [43mget_ipython[49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[43mrun_line_magic[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mmatplotlib[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[38;5;124;43m'[39;49m[38;5;124;43mwidget[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m      3[0m get_ipython()[38;5;241m.[39mrun_line_magic([38;5;124m'[39m[38;5;124mgui[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124masyncio[39m[38;5;124m'[39m)
[1;32m      5[0m [38;5;28;01mfrom[39;00m [38;5;21;01mai2_kit[39;00m[38;5;21;01m.[39;00m[38;5;21;01mfeat[39;00m [38;5;28;01mimport[39;00m catalysis [38;5;28;01mas[39;00m ai2cat

File [0;32m~/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/IPython/core/interactiveshell.py:2456[0m, in [0;36mInteractiveShell.run_line_magic[0;34m(self, magic_name, line, _stack_depth)[0m
[1;32m   2454[0m     kwargs[[38;5;124m'[39m[38;5;124mlocal_ns[39m[38;5;124m'[39m] [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mget_local_scope(stack_depth)
[1;32m   2455[0m [38;5;28;01mwith[39;00m [38;5;28mself[39m[38;5;241m.[39mbuiltin_trap:
[0;32m-> 2456[0m     result [38;5;241m=[39m [43mfn[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m   2458[0m [38;5;66;03m# The code below prevents the output from being displayed[39;00m
[1;32m   2459[0m [38;5;66;03m# when using magics with decorator @output_can_be_silenced[39;00m
[1;32m   2460[0m [38;5;66;03m# when the last Python token in the expression is a ';'.[39;00m
[1;32m   2461[0m [38;5;28;01mif[39;00m [38;5;28mgetattr[39m(fn, magic[38;5;241m.[39mMAGIC_OUTPUT_CAN_BE_SILENCED, [38;5;28;01mFalse[39;00m):

File [0;32m~/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/IPython/core/magics/pylab.py:99[0m, in [0;36mPylabMagics.matplotlib[0;34m(self, line)[0m
[1;32m     97[0m     [38;5;28mprint[39m([38;5;124m"[39m[38;5;124mAvailable matplotlib backends: [39m[38;5;132;01m%s[39;00m[38;5;124m"[39m [38;5;241m%[39m backends_list)
[1;32m     98[0m [38;5;28;01melse[39;00m:
[0;32m---> 99[0m     gui, backend [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mshell[49m[38;5;241;43m.[39;49m[43menable_matplotlib[49m[43m([49m[43margs[49m[38;5;241;43m.[39;49m[43mgui[49m[38;5;241;43m.[39;49m[43mlower[49m[43m([49m[43m)[49m[43m [49m[38;5;28;43;01mif[39;49;00m[43m [49m[38;5;28;43misinstance[39;49m[43m([49m[43margs[49m[38;5;241;43m.[39;49m[43mgui[49m[43m,[49m[43m [49m[38;5;28;43mstr[39;49m[43m)[49m[43m [49m[38;5;28;43;01melse[39;49;00m[43m [49m[43margs[49m[38;5;241;43m.[39;49m[43mgui[49m[43m)[49m
[1;32m    100[0m     [38;5;28mself[39m[38;5;241m.[39m_show_matplotlib_backend(args[38;5;241m.[39mgui, backend)

File [0;32m~/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3645[0m, in [0;36mInteractiveShell.enable_matplotlib[0;34m(self, gui)[0m
[1;32m   3641[0m         [38;5;28mprint[39m([38;5;124m'[39m[38;5;124mWarning: Cannot change to a different GUI toolkit: [39m[38;5;132;01m%s[39;00m[38;5;124m.[39m[38;5;124m'[39m
[1;32m   3642[0m                 [38;5;124m'[39m[38;5;124m Using [39m[38;5;132;01m%s[39;00m[38;5;124m instead.[39m[38;5;124m'[39m [38;5;241m%[39m (gui, [38;5;28mself[39m[38;5;241m.[39mpylab_gui_select))
[1;32m   3643[0m         gui, backend [38;5;241m=[39m pt[38;5;241m.[39mfind_gui_and_backend([38;5;28mself[39m[38;5;241m.[39mpylab_gui_select)
[0;32m-> 3645[0m [43mpt[49m[38;5;241;43m.[39;49m[43mactivate_matplotlib[49m[43m([49m[43mbackend[49m[43m)[49m
[1;32m   3646[0m configure_inline_support([38;5;28mself[39m, backend)
[1;32m   3648[0m [38;5;66;03m# Now we must activate the gui pylab wants to use, and fix %run to take[39;00m
[1;32m   3649[0m [38;5;66;03m# plot updates into account[39;00m

File [0;32m~/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/IPython/core/pylabtools.py:361[0m, in [0;36mactivate_matplotlib[0;34m(backend)[0m
[1;32m    356[0m matplotlib[38;5;241m.[39minteractive([38;5;28;01mTrue[39;00m)
[1;32m    358[0m [38;5;66;03m# Matplotlib had a bug where even switch_backend could not force[39;00m
[1;32m    359[0m [38;5;66;03m# the rcParam to update. This needs to be set *before* the module[39;00m
[1;32m    360[0m [38;5;66;03m# magic of switch_backend().[39;00m
[0;32m--> 361[0m [43mmatplotlib[49m[38;5;241;43m.[39;49m[43mrcParams[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43mbackend[39;49m[38;5;124;43m'[39;49m[43m][49m [38;5;241m=[39m backend
[1;32m    363[0m [38;5;66;03m# Due to circular imports, pyplot may be only partially initialised[39;00m
[1;32m    364[0m [38;5;66;03m# when this function runs.[39;00m
[1;32m    365[0m [38;5;66;03m# So avoid needing matplotlib attribute-lookup to access pyplot.[39;00m
[1;32m    366[0m [38;5;28;01mfrom[39;00m [38;5;21;01mmatplotlib[39;00m [38;5;28;01mimport[39;00m pyplot [38;5;28;01mas[39;00m plt

File [0;32m~/.cache/pypoetry/virtualenvs/ai2-kit-AqYCtgTs-py3.9/lib/python3.9/site-packages/matplotlib/__init__.py:738[0m, in [0;36mRcParams.__setitem__[0;34m(self, key, val)[0m
[1;32m    736[0m         cval [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mvalidate[key](val)
[1;32m    737[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mValueError[39;00m [38;5;28;01mas[39;00m ve:
[0;32m--> 738[0m         [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mKey [39m[38;5;132;01m{[39;00mkey[38;5;132;01m}[39;00m[38;5;124m: [39m[38;5;132;01m{[39;00mve[38;5;132;01m}[39;00m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m [38;5;28;01mNone[39;00m
[1;32m    739[0m     [38;5;28mself[39m[38;5;241m.[39m_set(key, cval)
[1;32m    740[0m [38;5;28;01mexcept[39;00m [38;5;167;01mKeyError[39;00m [38;5;28;01mas[39;00m err:

[0;31mValueError[0m: Key backend: 'module://ipympl.backend_nbagg' is not a valid value for backend; supported values are ['gtk3agg', 'gtk3cairo', 'gtk4agg', 'gtk4cairo', 'macosx', 'nbagg', 'notebook', 'qtagg', 'qtcairo', 'qt5agg', 'qt5cairo', 'tkagg', 'tkcairo', 'webagg', 'wx', 'wxagg', 'wxcairo', 'agg', 'cairo', 'pdf', 'pgf', 'ps', 'svg', 'template', 'inline']

